raft: processCommittedEntries: passed in group and fetched group objects do not match

via `TestRaftRemoveRace` in https://circleci.com/gh/tschottdorf/cockroach/1704

```
panic: passed in group and fetched group objects do not match
original:&{groupID:1 replicaID:9 committedTerm:6 leader:{NodeID:1 StoreID:1 ReplicaID:1} pending:map[] writing:false nodeIDs:[1 2 3]}
fetched:&{groupID:1 replicaID:10 committedTerm:6 leader:{NodeID:1 StoreID:1 ReplicaID:1} pending:map[] writing:false nodeIDs:[1 2 3]}


goroutine 6355 [running]:
github.com/cockroachdb/cockroach/multiraft.(*state).processCommittedEntry.func1.1()
	/go/src/github.com/cockroachdb/cockroach/multiraft/multiraft.go:1162 +0x32b
github.com/cockroachdb/cockroach/multiraft.(*state).start.func1()
	/go/src/github.com/cockroachdb/cockroach/multiraft/multiraft.go:668 +0x144f
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8206196e0, 0xc82061eb30)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:88 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x62
```

--- a/multiraft/multiraft.go
+++ b/multiraft/multiraft.go
@@ -1158,8 +1158,8 @@ func (s *state) processCommittedEntry(groupID roachpb.RangeID, g *group, entry r
 					if !ok {
 						log.Infof("group %d no longer exists, aborting configuration change", groupID)
 					} else if gInner != g {
-						panic(fmt.Sprintf("passed in group and fetched group objects do not match\noriginal:%+v\nfetched:%+v\n",
-							g, gInner))
+						log.Infof("passed in group and fetched group objects do not match\noriginal:%+v\nfetched:%+v\n, aborting configuration change",
+							g, gInner)
 					} else if err == nil {
 						if log.V(3) {
 							log.Infof("node %v applying configuration change %v", s.nodeID, cc)

--- a/storage/client_raft_test.go
+++ b/storage/client_raft_test.go
@@ -1131,7 +1131,6 @@ func TestRaftAfterRemoveRange(t *testing.T) {
 // number of repetitions adds an unacceptable amount of test runtime).
 func TestRaftRemoveRace(t *testing.T) {
 	defer leaktest.AfterTest(t)
-	t.Skip("#3187, #3178 and others; very flaky since PR #3175")
 	mtc := startMultiTestContext(t, 3)
 	defer mtc.Stop()
 

